easybuild/easyblocks/p/pytorch.py

```diff
@@ -152,9 +152,23 @@ def configure_step(self):
         if get_software_root('imkl'):
             options.append('BLAS=MKL')
             options.append('INTEL_MKL_DIR=$MKLROOT')
-        else:
-            # This is what PyTorch defaults to if no MKL is found. Make this explicit here
+        elif LooseVersion(self.version) >= LooseVersion('1.9.0') and get_software_root('BLIS'):
+            options.append('BLAS=BLIS')
+            options.append('BLIS_HOME=' + get_software_root('BLIS'))
+            options.append('USE_MKLDNN_CBLAS=ON')
+        elif get_software_root('OpenBLAS'):
+            # This is what PyTorch defaults to if no MKL is found.
+            # Make this explicit here to avoid it finding MKL from the system
             options.append('BLAS=Eigen')
```

Comment:
maybe stupid question, but why not just set BLAS=OpenBLAS here?

Reply:
Because PyTorch uses the former cafe code and for some reasons seems to prefer using that with Eigen, which will use a BLAS as a backend.  PyTorch itself uses BLAS directly (at least this is what I gathered from various issues)
To not change our current behavior and not deviate from the default PyTorch builds (for the no-mkl case) I chose to keep this as Eigen (same as PyTorch does)

And yes it is a mess: https://github.com/pytorch/pytorch/issues/59218